### ISSUE TITLE
linter: unexpected detection with jsdoc

### ISSUE THREAD

**Issue by Mister-Hope** [oxc-project/oxc#19139]
### What happened?

When you add a jsdoc to a function assignment, jsdoc should only check the current function jsdoc, detecting whether it's tag name is valid and whether it has required `@params` and `@returns`.

However, now jsdocs complains about every function expression inside such a function expression, which should not be the expected behavior. Here are some examples:

Case 1: A markdown plugin

```ts
/**
 * Component fence support plugin
 *
 * 组件代码块支持插件
 *
 * @param md - MarkdownIt instance / MarkdownIt 实例
 */
export const component: PluginSimple = (md) => {
  // Handle ```component  blocks
  const { fence } = md.renderer.rules

  md.renderer.rules.fence = (...args): string => {
    const [tokens, index] = args
    const { info } = tokens[index]

    const [fenceName, componentName] = info.split(' ', 2)

    if (fenceName === 'component') {
      const render = getComponentRender(componentName)

      return render(...args)
    }

    return fence!(...args)
  }
}
```

The above code already has the required jsdoc information; however, since a function is found inside, JSDoc complains:

<img width="1593" height="819" alt="Image" src="https://github.com/user-attachments/assets/4c975fd7-b9c1-4635-b644-df001a4634a8" />

This behavior is NOT acceptable in any codebase. Another case here:

<img width="1206" height="1023" alt="Image" src="https://github.com/user-attachments/assets/fa0743eb-127a-436e-9c2a-a4af17672d3c" />

(the outer function has a jsdoc, so this assignement requires a jsdoc, this also makes no sence)

The outer function:

```ts
/**
 * Use "additionalData" to make config and palette modules available in SCSS
 *
 * 使用 "additionalData" 使配置和调色板模块在 SCSS 中可用
 *
 * @description Injects SCSS config and palette modules into bundler configuration using additionalData option.
 *
 * 使用 additionalData 选项将 SCSS 配置和调色板模块注入到打包器配置中。
 *
 * @param config - VuePress bundler configuration / VuePress 打包器配置
 * @param app - VuePress app instance / VuePress 应用实例
 * @param id - Module identifier for the palette / 调色板的模块标识符
 */
export const injectScssConfigModule = (
  config: unknown,
  app: App,
  id: string,
): void => {
// ....
```

Also, I am thinking whether we can make improvements by introducing type information to jsdoc.

A example can be a shiki transformer:

<img width="1431" height="1026" alt="Image" src="https://github.com/user-attachments/assets/ab6b6838-bd4b-4494-96ac-d5faf2fd0e20" />

We already added enough information about what the transformers do. However, oxlint still complains about missing param definition, which should be shiki's work and should be done with its `ShikiTransformer` type, it's not our job to add such in our code:

<img width="1182" height="339" alt="Image" src="https://github.com/user-attachments/assets/28e32269-987c-4c32-8a8c-5713bcebfa9d" />

If type information is provided, then probably jsdoc can understand the object is typed by `ShikiTransformer`, so `@param` and `@returns` is not needed here if users do not manually add a jsdoc in the function property inside object. I haven't read the jsdoc plugin, maybe type information is not required, as jsdoc plugin can understand the object is typed just from AST.

I hope the first one can be fixed, and the second improvement can be discussed somewhere.

I believe most developers will choose to toogle jsdoc plugin off like me with these issues. We can not throw `// oxlint-disable jsdoc/xxx` everywhere in our projects.

**Comment by CraigglesO** [comment]
Can confirm it fails very often.

**Comment by Mister-Hope** [comment]
@camc314 I did not see any fix about the shikiTransformer example, nor related test cases, shall I open a new issue to mention it, or making some discussion just here and reopen it?

**Comment by camc314** [comment]
Thanks @Mister-Hope good catcch, i;ve put up another PR to fix
